lib/sample_uploader/sample_uploaderImpl.py

```diff
@@ -642,6 +644,90 @@ def batch_link_samples(self, ctx, params):
+           structure: parameter "workspace_name" of String, parameter
+           "workspace_id" of String, parameter "out_sample_set_name" of
+           String, parameter "sample_set_ref" of list of String, parameter
+           "filter_conditions" of list of type "FilterCondition" (Filter
+           SampleSets) -> structure: parameter "column" of String, parameter
+           "comparison" of String, parameter "value" of String, parameter
+           "condition" of String
+        :returns: instance of type "FilterSampleSetsOutput" -> structure:
+           parameter "report_name" of String, parameter "report_ref" of String
+        """
+        # ctx is the context object
+        # return variables are: output
+        #BEGIN filter_samplesets
+        samples = []
+        for sample_set in self.dfu.get_objects({'object_refs': params['sample_set_ref']})['data']:
+            samples.extend(sample_set['data']['samples'])
+        sample_ids = [{'id': sample['id'], 'version':sample['version']} for sample in samples]
+
+        filter_conditions = []
+        for condition in params['filter_conditions']:
+            values = []
+            if (condition['comparison'].strip().lower() not in ["in", "not in"]):
+                values = [condition['value']]
+            else:
+                values = [v.strip() for v in condition['value'].split(", ")]
+            filter_conditions.append({
+                'metadata_field': _upload_key_format(condition['column']),
+                'comparison_operator': condition['comparison'],
+                'metadata_values': values,
+                'logical_operator': condition['condition']
+            })
+
+        sample_search_api_request = {
+            'sample_ids': sample_ids,
+            'filter_conditions': filter_conditions
+        }
+
+        sample_search_api_response = sample_search_api(
+            url=self.sw_url, service_ver='dev').filter_samples(sample_search_api_request)
+
+        samples_to_keep = set(sample_id['id']
+                              for sample_id in sample_search_api_response['sample_ids'])
+
+        sample_set = {
+            'samples': [sample for sample in samples if sample['id'] in samples_to_keep],
+            "description": "Generated with the Sample Set Editor"
+        }
+
+        obj_info = self.dfu.save_objects({
+            'id': params['workspace_id'],
+            'objects': [{
+                "name": params['out_sample_set_name'],
+                "type": "KBaseSets.SampleSet",
+                "data": sample_set
+            }]
+        })
+
+        report_client = KBaseReport(self.callback_url)
+        report_info = report_client.create_extended_report({
+            'objects_created': [
+                {
+                    'ref': "/".join([str(info[6]), str(info[0]), str(info[4])])
+                } for info in obj_info
+            ],
+            'message': f"SampleSet object named \"{params['out_sample_set_name']}\" created.",
```

Comment:
Maybe also include some info about the filtering parameters used in the report.

Reply:
👌